python/shotgun_model/shotgun_model.py

```diff
@@ -841,6 +865,62 @@ def _load_external_data(self):
         """
         pass
 
+    def _get_additional_columns(self, item, is_leaf, columns):
```

Comment:
If I am reading this correctly, we are basically only caching column zero, and this column keeps a full shotgun data chunk representing all the column data. Then at runtime, we "unpack" this by executing `_get_additional_columns()` for each individual row and non-primary column. 

I feel this approach is elegant because it allows for columns to be pretty dynamic in the model, yet at the same time the same cache can be shared and re-used even as columns are treated differently in different cases. Maybe it's premature, but should we be worried about performance? Maybe it's nothing to worry about, but suddenly a majority of the items are created dynamically in python rather than deserialized in directly from disk.